src/components/dashboard/menu-table/dropdown.js

```diff
@@ -25,22 +27,32 @@ const Categories = styled.div`
     }
 `
 
-class Dropdown extends React.Component {
+class CategoryDropdown extends React.Component {
     constructor(props) {
         super(props);
 
         this.state = {
             displayList: false,
             data: [],
             currentSelection: '',
+            categoryOptions: [],
+            categoryNames: [],
+            currCategory: '',
         }
 
-        this.showDropdown = this.showDropdown.bind(this)
-        this.hideDropdown = this.hideDropdown.bind(this)
-
         Client.getMenu(props.menuId).then((res) => {
-            console.log(res.data)
+            console.log("client ", res.data)
             this.state.data = res.data.Categories
+
+            for (let i = 0; i < res.data.Categories.length; i++) {
+                this.state.categoryNames[i] = res.data.Categories[i].name;
+            }
+            this.state.categoryOptions = _.map(this.state.categoryNames, (category, index) => ({
+                key: index + 1,
+                text: category,
+                value: category
+            }));
```

Comment:
Please use set state to modify any object in this.state (like merging them into line 55). If you could fix the other ones in the file that'll be great :)